bson/doc.go

```diff
@@ -103,9 +103,13 @@
 // to configure behavior:
 //
 //     1. omitempty: If the omitempty struct tag is specified on a field, the field will not be marshalled if it is set to
-//     the zero value. By default, a struct field is only considered empty if the field's type implements the Zeroer
-//     interface and the IsZero method returns true. Struct fields of types that do not implement Zeroer are always
-//     marshalled as embedded documents. This tag should be used for all slice and map values.
+//     the zero value. Fields with language primitive types such as integers, booleans, and strings are considered empty if
+//     their value is equal to the zero value for the type (i.e. 0 for integers, false for booleans, and "" for strings).
+//     Slices, maps, and arrays are considered empty if they are of length zero. Interfaces and pointers are considered
```

Comment:
No change needed, but "length zero" implies nil slice / map/ array is considered zero, right? Though I assume any user would infer that if the length of a nil slice is 0.

Reply:
Yes. I wrote "length zero" because that covers both nil slices and non-nil empty slices. Also `len(slice)` returns 0 if `slice` is nil.